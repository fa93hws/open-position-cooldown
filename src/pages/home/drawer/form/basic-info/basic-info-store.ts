import type { InputStore } from '@ui/text-input/input-store';
import type { DatePickerStore } from './date-picker/date-picker-store';

export class BasicInfoStore {
  private readonly nameStore: InputStore;

  private readonly priceStore: InputStore;

  private readonly codeStore: InputStore;

  private readonly marketStore: InputStore;

  private readonly datePickerStore: DatePickerStore;

  constructor(params: {
    nameStore: InputStore;
    priceStore: InputStore;
    codeStore: InputStore;
    marketStore: InputStore;
    datePickerStore: DatePickerStore;
  }) {
    this.nameStore = params.nameStore;
    this.priceStore = params.priceStore;
    this.codeStore = params.codeStore;
    this.marketStore = params.marketStore;
    this.datePickerStore = params.datePickerStore;
  }

  get name() {
    return this.nameStore.value;
  }

  get price() {
    return this.priceStore.value;
  }

  get priceAsNum() {
    return parseFloat(this.price);
  }

  get code() {
    return this.codeStore.value;
  }

  get market() {
    return this.marketStore.value;
  }

  get hasError() {
    return (
      this.nameStore.hasError ||
      this.priceStore.hasError ||
      this.codeStore.hasError ||
      this.marketStore.hasError
    );
  }

  get date() {
    return this.datePickerStore.asDateString();
  }

  startValidate() {
    this.nameStore.startValidate();
    this.priceStore.startValidate();
    this.codeStore.startValidate();
    this.marketStore.startValidate();
  }

  reset() {
    this.nameStore.reset();
    this.priceStore.reset();
    this.codeStore.reset();
    this.marketStore.reset();
    this.datePickerStore.reset();
  }
}
