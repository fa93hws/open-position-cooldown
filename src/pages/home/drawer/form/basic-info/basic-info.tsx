import * as React from 'react';
import { observer } from 'mobx-react';
import { Box, Typography } from '@material-ui/core';
import { createInput } from '@ui/text-input/input';
import { nonEmpty, mustBeNumber } from '@ui/text-input/validator';
import { BasicInfoStore } from './basic-info-store';
import { createDatePicker } from './date-picker/date-picker';

export function createBasicInfo(): [React.ComponentType, BasicInfoStore] {
  const [NameInput, nameStore] = createInput([nonEmpty]);
  const [CodeInput, codeStore] = createInput([nonEmpty]);
  const [PriceInput, priceStore] = createInput([nonEmpty, mustBeNumber]);
  const [MarketInput, marketStore] = createInput([nonEmpty]);
  const [DatePickerInput, datePickerStore] = createDatePicker();
  const store = new BasicInfoStore({
    nameStore,
    codeStore,
    priceStore,
    marketStore,
    datePickerStore,
  });

  const Component = observer(() => (
    <>
      <Box marginBottom={1}>
        <NameInput label="股票名" />
      </Box>
      <Box marginBottom={1}>
        <CodeInput label="代码" />
      </Box>
      <Box marginBottom={1}>
        <PriceInput label="价格" />
      </Box>
      <Box marginBottom={1}>
        <MarketInput label="市场" />
      </Box>
      <Box
        marginBottom={2}
        display="flex"
        alignItems="center"
        justifyContent="space-between"
      >
        <Typography variant="body1" component="span">
          日期:
        </Typography>
        <DatePickerInput />
      </Box>
    </>
  ));
  return [Component, store];
}
