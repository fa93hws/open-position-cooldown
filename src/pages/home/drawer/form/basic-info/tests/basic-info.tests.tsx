import * as React from 'react';

import { createBasicInfo } from '../basic-info';

describe('BasicInfo', () => {
  it('renders the following', () => {
    const [BasicInfo, store] = createBasicInfo();
    store.datePickerStore.setDate(new Date('1999-02-04'));
    expect(<BasicInfo />).toMatchRenderedSnapshot();
  });
});
