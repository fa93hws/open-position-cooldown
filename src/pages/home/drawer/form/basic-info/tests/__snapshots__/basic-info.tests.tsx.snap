// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BasicInfo renders the following 1`] = `
Array [
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
      >
        股票名
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-4"
          >
            <span>
              股票名
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  <div
    class="MuiBox-root MuiBox-root-6"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
      >
        代码
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-4"
          >
            <span>
              代码
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  <div
    class="MuiBox-root MuiBox-root-7"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
      >
        价格
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-4"
          >
            <span>
              价格
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  <div
    class="MuiBox-root MuiBox-root-8"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
      >
        市场
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-4"
          >
            <span>
              市场
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
  <div
    class="MuiBox-root MuiBox-root-9"
  >
    <span
      class="MuiTypography-root MuiTypography-body1"
    >
      日期:
    </span>
    <div
      class="MuiFormControl-root MuiTextField-root"
      format="yyyy-MM-dd"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          readonly=""
          type="text"
          value="2020-10-18"
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
          style="padding-left:8px"
        >
          <legend
            class="PrivateNotchedOutline-legend-3"
            style="width:0.01px"
          >
            <span>
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>,
]
`;
